### DISCUSSION TITLE
Difference between ripgrep, plocate and tools like everything, flow-launcher, launchy, Listary,...

### DISCUSSION THREAD

**Discussion by Ironwally** [BurntSushi/ripgrep#2835] [Q&A] [answered]
Hi! 
If've been looking for a software to quickly search my computer for files, directories, anything really.
Until now I have been using Listary, but have become suspicious concearnig private data, etc.
While searching I found ripgrep.

My question is:
1. What is the difference between repgrep and plocate? 
These tools seem quite similar, just that plocate only searches names (and therefore might be optimised better for that usecase?)
2. Can this tool be used in a similar way to Everything or other file search engines

Thank you in advance.
I'm sorry if this question seems stupid, I could not find any information on this subject here or on the internet.

Best regards!

**Comment by BurntSushi** [comment, marked as answer]
Hi. In the future, when asking for comparisons between tools, **please please please** provide links to those tools. Some of the tools you mention I've never heard of. So I have to do a search and hope that the project homepage I land on is the same one you had in mind.

> 1. What is the difference between repgrep and plocate?
>        These tools seem quite similar, just that plocate only searches names (and therefore might be optimised better for that usecase?)

Assuming you mean [`plocate`](https://plocate.sesse.net/), then it pretty clearly advertises itself as a `locate` derivative. As you say, `locate` is only designed to search file names. ripgrep searches the contents of files and not file names at all. While both are "search tools," they are quite different. There really isn't any other similarity. So I personally would not call them "quite similar."

> Can this tool be used in a similar way to Everything or other file search engines

I'm assuming you mean [Everything](https://www.voidtools.com/support/everything/). If so, I've never used the tool. A quick skim of its documentation suggests it is a GUI for searching the names of files. Once again, ripgrep does not search the names of files. It only searches the contents of files.

Like `locate` based tools, Everything also seems to have an indexing step. But again, I don't see anything to suggest that it searches the contents of files.

> I'm sorry if this question seems stupid, I could not find any information on this subject here or on the internet.

It's a fine question. But I think the answer basically boils down to the fact that `locate` and Everything seem to be tools for searching the names of files, and where both are powered by a pre-constructed index. ripgrep, on the other hand, does not search the names of files at all. It only searches the contents of files. And ripgrep does not support having a pre-constructed index. It exhaustively searches whatever input you give it. So if you're looking to search an entire hard drive, depending on how big it is, ripgrep might take quite a bit of time.

ripgrep isn't really a tool that is intended to provide a complete end user "search engine" experience. It's a tool that, at the highest level, is designed to search plain text files line-by-line, and print lines that match the query. On top of that, it has some conveniences optimized for programmers working in code repositories, like automatically respecting your `.gitignore` and ignoring hidden files and directories by default. (That's about as close to "search engine" as ripgrep gets.)

**Reply by Ironwally** [reply]
Ah, ok.
Yes, my question was really about the possibility to use ripgrep to search hard drives and if it used a pre-constructed index.
Thanks for explaining that.
 
And also thanks for the feedback, I'll do that :).
